mapie/quantile_regression.py

```diff
@@ -363,16 +369,41 @@ def fit(  # type: ignore
         alpha = self._check_alpha(self.alpha)
         self.cv = self._check_cv(cast(str, self.cv))
         X, y = indexable(X, y)
+        if X_calib is None or y_calib is None:
+            if sample_weight is not None:
+                (
+                    X_train,
+                    X_calib,
+                    y_train,
+                    y_calib,
+                    sample_weight_train,
+                    _,
+                ) = train_test_split(
+                    X,
+                    y,
+                    sample_weight,
+                    test_size=calib_size
+                )
+            else:
+                X_train, X_calib, y_train, y_calib = train_test_split(
+                    X,
+                    y,
+                    test_size=calib_size
+                )
+                sample_weight_train = sample_weight
+        else:
+            X_train, y_train, sample_weight_train = X, y, sample_weight
```

Comment:
I would move this block in a dedicated hidden method.

Reply:
done